sgkit_vcf/tests/test_vcf_reader.py

```diff
@@ -0,0 +1,188 @@
+import numpy as np
+import pytest
+import xarray as xr
+from numpy.testing import assert_array_equal
+
+from sgkit_vcf import partition_into_regions, vcf_to_zarr
+
+
+def test_vcf_to_zarr__small_vcf(shared_datadir):
+    path = shared_datadir / "sample.vcf.gz"
+    output = "vcf.zarr"
+
+    vcf_to_zarr(path, output, chunk_length=5, chunk_width=2)
+    ds = xr.open_zarr(output)  # type: ignore[no-untyped-call]
+
+    assert ds.attrs["contigs"] == ["19", "20", "X"]
+    assert_array_equal(ds["variant_contig"], [0, 0, 1, 1, 1, 1, 1, 1, 2])
+    assert_array_equal(
+        ds["variant_position"],
+        [111, 112, 14370, 17330, 1110696, 1230237, 1234567, 1235237, 10],
+    )
+    assert_array_equal(
```

Comment:
Worth testing that the fixed length string dtype is correct too?

Reply:
https://github.com/pystatgen/sgkit-vcf/pull/1/commits/90a59523c74e810f5be6def4fa989ffd44c3950d